haystack/components/preprocessors/document_splitter.py

```diff
@@ -61,18 +62,24 @@ def __init__(
         :param split_overlap: The number of overlapping units for each split.
         :param split_threshold: The minimum number of units per split. If a split has fewer units
             than the threshold, it's attached to the previous split.
+        :param splitting_function: Necessary when `split_by` is set to "function".
+            This is a function which must accept a single `str` as input and return a `list` of `str` as output,
+            representing the chunks after splitting.
         """
 
         self.split_by = split_by
-        if split_by not in ["word", "sentence", "page", "passage"]:
+        if split_by not in ["function", "page", "passage", "sentence", "word"]:
             raise ValueError("split_by must be one of 'word', 'sentence', 'page' or 'passage'.")
```

Comment:
Ah, tiny nitpick here @GivAlz we need to add 'function' in this list here. Let's fix this and integrate. Don't forget to pull branch first as I merged main into it 🙏